### ISSUE TITLE
Content pages can be rendered event if the extension is not enabled/installed

### ISSUE THREAD

**Issue by nitriques** [symphonycms/symphony-2#2367]
I really do not think this is a feature, hence it's a bug.

Is it a security issue/concern ?

**Comment by brendo** [comment]
It's definitely not a feature, that's for sure.

It's the `Administration` class that should get a bit smarter and check if an Extension is installed during the route resolution (somewhere in [`getPageCallback()`](https://github.com/symphonycms/symphony-2/blob/master/symphony/lib/core/class.administration.php#L309) I'd imagine). The `ExtensionManager` class is pretty lazy here, it doesn't really have a suitable method to call at this stage. While we can call `fetchStatus`, that unfortunately requires an associative array with handle and version keys. Alternatively we can call `listInstalledHandles` and check the array. Both will work, but aren't particularly nice to use. Probably worthwhile introducing a new accessor for `isInstalled($handle)`.

**Comment by nitriques** [comment]
> Probably worthwhile introducing a new accessor for isInstalled($handle).

Will do. Thanks.

**Comment by nitriques** [comment]
I can't decide which one we should use....

Version 1, strait DB call..

```php
    /**
     * Returns true if extension $handle is installed
     * @since Symphony 2.6.0
     * @return boolean
     */
    public static function isInstalled($handle)
    {
        if (Symphony::Database()->isConnected()) {
            $handle = MySQL::cleanValue($handle);
            $count = Symphony::Database()->fetchCol(
                'c',
                "SELECT count(`id`) as `c` FROM `tbl_extensions` WHERE `status` = 'enabled'
                 AND `name` = '$handle'"
            );
            return intval($count) == 1;
        }
        return false;
    }
```

Works ok, simple enough, makes sure we have the latest value...

---

Version 2

```php
    /**
     * Returns true if extension $handle is installed
     * @since Symphony 2.6.0
     *
     * @return boolean
     */
    public static function isInstalled($handle)
    {
        return in_array($handle, self::listInstalledHandles());
    }
```

Even simpler, but load extra data in some cases (returning ALL extensions name)

**Comment by nitriques** [comment]
I woul also add

```php
$callback['extension'] = $extension_name;
```

**Comment by nitriques** [comment]
Here's a temporary commit for review https://github.com/DeuxHuitHuit/symphony-2/commit/ae6f51b94ba5afda672e25529f13176fb064b3cf

**Comment by brendo** [comment]
This will not make 2.6.0 final.

**Comment by nitriques** [comment]
No prob. Could be 2.7 ? Or we going strait to 3.0 ?

**Comment by nitriques** [comment]
Also, which version do you prefer ? :smile:

**Comment by nitriques** [comment]
Merged as 7406725ac8
